src/NUnitFramework/framework/Constraints/ConstraintExpression.cs

```diff
@@ -427,6 +427,49 @@ public EqualConstraint EqualTo(object? expected)
             return Append(new EqualConstraint(expected));
         }
 
+        /// <summary>
+        /// Returns a constraint that tests two strings for equality
+        /// </summary>
+        public EqualStringConstraint EqualTo(string? expected)
+        {
+            return Append(new EqualStringConstraint(expected));
+        }
+
+        /// <summary>
+        /// Returns a constraint that tests two date time offset instances for equality
+        /// </summary>
+        public EqualDateTimeOffsetConstraint EqualTo(DateTimeOffset expected)
+        {
+            return Append(new EqualDateTimeOffsetConstraint(expected));
+        }
+
+        /// <summary>
+        /// Returns a constraint that tests two date time instances for equality
+        /// </summary>
+        public EqualTimeBaseConstraint<DateTime> EqualTo(DateTime expected)
+        {
+            return Append(new EqualTimeBaseConstraint<DateTime>(expected, x => x.Ticks));
+        }
+
+        /// <summary>
+        /// Returns a constraint that tests two timespan instances for equality
+        /// </summary>
+        public EqualTimeBaseConstraint<TimeSpan> EqualTo(TimeSpan expected)
+        {
+            return Append(new EqualTimeBaseConstraint<TimeSpan>(expected, x => x.Ticks));
+        }
+
+        /// <summary>
+        /// Returns a constraint that tests two numbers for equality
+        /// </summary>
+#pragma warning disable CS3024 // Constraint type is not CLS-compliant
+        public EqualNumericConstraint<T> EqualTo<T>(T expected)
+            where T : unmanaged, IConvertible, IEquatable<T>
+        {
```

Comment:
is this generic constraint `where T : unmanaged` meant to restrict to value types or is it really also for other unmanaged types ?  If the first, any reason not to use struct ?